bundles/org.openhab.binding.amberelectric/src/main/java/org/openhab/binding/amberelectric/internal/AmberElectricHandler.java

```diff
@@ -0,0 +1,152 @@
+
+    private long refreshInterval;
+    private String apiKey = "";
+    private String nmi = "";
+    private String siteID = "";
+
+    private @NonNullByDefault({}) AmberElectricConfiguration config;
+    private @NonNullByDefault({}) AmberElectricWebTargets webTargets;
+    private @Nullable ScheduledFuture<?> pollFuture;
+
+    public AmberElectricHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.warn("This binding is read only");
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(AmberElectricConfiguration.class);
+        if (config.apiKey.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error.no-api-key");
+        } else {
+            webTargets = new AmberElectricWebTargets();
+            updateStatus(ThingStatus.UNKNOWN);
+            refreshInterval = config.refresh;
+            nmi = config.nmi;
+            apiKey = config.apiKey;
+
+            schedulePoll();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        ScheduledFuture<?> pollFuture = this.pollFuture;
+        logger.debug("Scheduling poll every {} s", refreshInterval);
+        this.pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 0, refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private void poll() {
+        try {
+            logger.debug("Polling for state");
+            pollStatus();
+        } catch (IOException e) {
+            logger.debug("Could not connect to AmberAPI", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (RuntimeException e) {
+            logger.warn("Unexpected error connecting to AmberAPI", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void stopPoll() {
+        final Future<?> future = pollFuture;
+        if (future != null) {
+            future.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private void pollStatus() throws IOException {
+
+        try {
+            if (siteID.isEmpty()) {
+                Sites sites = webTargets.getSites(apiKey, nmi);
+                // add error handling
+                siteID = sites.siteid;
+                logger.debug("Detected amber siteid is {}, for nmi {}", sites.siteid, sites.nmi);
+            }
+
+            CurrentPrices currentPrices = webTargets.getCurrentPrices(siteID, apiKey);
+            updateStatus(ThingStatus.ONLINE);
+            updateState(AmberElectricBindingConstants.CHANNEL_AMBERELECTRIC_ELECPRICE,
+                    new QuantityType<>(currentPrices.elecPerKwh / 100, CurrencyUnits.BASE_ENERGY_PRICE));
```

Comment:
I think you will lose the currency unit this way? You might want to explicitly provide AUD/kWh, so the framework will be able to convert to desired currency - for example through [Freecurrency binding](https://next.openhab.org/addons/bindings/freecurrency/).

For an example, see also:
https://github.com/openhab/openhab-addons/blob/3ff89f82245c0b91e8d1d9ca4e18fd8e30512081/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/handler/EnergiDataServiceHandler.java#L489-L513

Reply:
thanks,  the code above gave me DEF/kWh until I set AUD in OpenHAB. I would prefer to set AUD, I just couldn't work out how to do it :) I'll investigate today.

Reply:
For reference: #16085

Reply:
working thru this now,  but temporarily have a "No thing types can be added with this binding." error adding a new Thing, so I have a typo/mismatch somewhere.

Reply:
OK fixed my stupid typos, and also  correctly (I think)  implemented  AUD/kWh. Thanks so much for the useful pointers, I had reviewed [EnergiDataServiceHandler.java](https://github.com/openhab/openhab-addons/blob/3ff89f82245c0b91e8d1d9ca4e18fd8e30512081/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/handler/EnergiDataServiceHandler.java#L489-L513) previously, but I was confused thinking CURRENCY_DKK was a Java constant - once I reviewed more of  the code,  the solution was obvious :)